testConfig.py

```diff
@@ -36,6 +37,27 @@
 T1 = TypeVar("T1")
 
 
+def get_resource_name() -> str:
+    try:
+        cmd = ["oc", "get", "network-attachment-definition", "-o", "yaml"]
+        nad_yaml = subprocess.check_output(cmd)
+    except subprocess.CalledProcessError as e:
+        logger.info(
+            f"Get network-attachment-defintion command failed with exit status {e.returncode}"
+        )
+        return "openshift.io/dpu"
```

Comment:
Note, that ocp-tft uses `kubectl` instead of `oc`. Presumably, to work with other kubernetes variants aside openshift. Can this be fetched via kubectl?

Also, there is `host.Host.run()` which abstracts calling processes. It should be used instead of `subprocess.check_output()`, because most notably it provides useful debug logging and error handling. Further, there is `K8sClient.oc*()`, which wraps `host.Host.run()`. That latter is the one that should be used here, because it exists exactly for this purpose. The intent is that `K8sClient.oc*()` is better than what you could reasonably re-implement here (whether it succeeds at that, you are welcome to question).

IOW, I think we should call here `data = k8sClient.oc_get("network-attachment-definition")`.

If you require `oc` command instead of `kubectl`, then `k8sClient.oc*()` should get an improvement to support that, and then call `k8sClient.oc_get("network-attachment-definition", oc_cmd="oc")`.

Reply:
okay I can use k8sclient.